numpy/lib/function_base.py

```diff
@@ -4154,32 +4155,87 @@ def _median(a, axis=None, out=None, overwrite_input=False):
         return mean(part[indexer], axis=axis, out=out)
 
 
+def _valid_quantile(q):
+    # avoid expensive reductions, relevant for arrays with < O(1000) elements
+    if q.ndim == 1 and q.size < 10:
+        for i in range(q.size):
+            if q[i] < 0.0 or q[i] > 1.0:
+                return False
+    else:
+        # faster than any()
+        if np.count_nonzero(q < 0.0) or np.count_nonzero(q > 1.0):
```

Comment:
These micro-optimizations don't seem worth it and don't help with code clarity.

Reply:
Indeed - however they were there before too, so I thought it better to leave them

Reply:
Trying to avoid ufunc call overhead everywhere would make a terrible mess. At some point we should make ufunc calls less expensive.